modules/Router.js

```diff
@@ -82,7 +84,10 @@ const Router = React.createClass({
       history,
       createRoutes(routes || children)
     )
+
     const router = createRouterObject(history, transitionManager)
+    router.location = null
+    router.params = null
```

Comment:
I just tried this but this doesn’t work for the initial render.
`setState` called inside `componentWillMount` will not cause `componentWillUpdate` to run.